api/app.py

```diff
@@ -0,0 +1,96 @@
+import connexion
+import os
+import json
+import logging
+from jupyterhub_singleuser_profiles.profiles import SingleuserProfiles
+
+from functools import wraps
+from urllib.parse import quote
+
+from flask import Flask
+from flask import redirect
+from flask import request
+from flask import Response
+
+from jupyterhub.services.auth import HubAuth
+
+_PROFILES = SingleuserProfiles(verify_ssl=False)
+_PROFILES.load_profiles()
+_PATH = "/opt/app-root/lib64/python3.6/site-packages"
+
+_LOGGER = logging.getLogger(__name__)
+
+prefix = os.environ.get('JUPYTERHUB_SERVICE_PREFIX', '/')
+
+auth = HubAuth(api_token=os.environ['JUPYTERHUB_API_TOKEN'], cache_max_age=60)
+
+app = Flask(__name__)
+
+def authenticated(f):
+    """Decorator for authenticating with the Hub"""
+
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        cookie = request.cookies.get(auth.cookie_name)
+        token = request.headers.get(auth.auth_header_name)
+        for_user = connexion.request.headers['For-User']
+        if not for_user :
+            for_user = 'null'
+        if cookie:
+            user = auth.user_for_cookie(cookie)
+        elif token:
+            user = auth.user_for_token(token)
+        else:
+            user = None
+        if for_user != 'null' and user['admin']:
```

Comment:
This can be just `if for_user and user.get('admin')` (I am not sure if the key `admin` is there always, probably yes, but in case we want to be on the safe side..